src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -0,0 +1,363 @@
+
+        private final String queueName;
+        private long size;
+        private long failures;
+        private long backpressureTime;
+        private long slowdownTime;
+
+        QueueStatistic(String queueName) {
+            this.queueName = queueName;
+        }
+
+        void setSize(Long size) {
+            if (size != null && size >= 0) {
+                this.size = size;
+                return;
+            }
+            this.size = 0;
+        }
+
+        void setFailures(Long failures) {
+            if (failures != null && failures >= 0) {
+                this.failures = failures;
+                return;
+            }
+            this.failures = 0;
+        }
+
+        void setBackpressureTime(Long backpressureTime) {
+            if (backpressureTime != null && backpressureTime >= 0) {
+                this.backpressureTime = backpressureTime;
+                return;
+            }
+            this.backpressureTime = 0;
+        }
+
+        void setSlowdownTime(Long slowdownTime) {
+            if (slowdownTime != null && slowdownTime >= 0) {
+                this.slowdownTime = slowdownTime;
+                return;
+            }
+            this.slowdownTime = 0;
+        }
+
+        JsonObject getAsJsonObject() {
+            return new JsonObject()
+                .put(MONITOR_QUEUE_NAME, queueName)
+                .put(MONITOR_QUEUE_SIZE, size)
+                .put(STATISTIC_QUEUE_FAILURES, failures)
+                .put(STATISTIC_QUEUE_BACKPRESSURE, backpressureTime)
+                .put(STATISTIC_QUEUE_SLOWDOWN, slowdownTime);
+        }
+    }
+
+
+    /**
+     * Retrieve the queue statistics for the requested queues.
+     *
+     * Note: This operation does interact with Redis to retrieve the values for the statistics
+     * for all queues requested (independent of the redisques instance for which the queues are
+     * registered). Therefore this method must be used with care and not be called too often!
+     *
+     * @param event  The event on which we will answer finally
+     * @param queues The queues for which we are interested in the statistics
+     */
+    public void getQueueStatistics(Message<JsonObject> event, final List<String> queues) {
+        if (queues == null || queues.isEmpty()) {
+            log.debug("Queue statistics evaluation with empty queues, returning empty result");
+            event.reply(new JsonObject().put(STATUS, OK).put(RedisquesAPI.QUEUES, new JsonArray()));
+            return;
+        }
+        // first retrieve all queue sizes from the queues itself by help of the LUA script
+        // which returns for each requested queueName the corresponding queue size.
+        // Note: If a queue doesn't exists, it will anyway return 0 for the same, therefore
+        //       the size of the returned queues must be equal in any case and has the same
+        //       order as the requested queues.
+        List<String> queueKeys = new ArrayList<>();
```

Comment:
Will do due to performance hopes, not because I find it "nicer"

Reply:
> Will do due to performance hopes

I guess this will trigger the opposite ;) But keep going. I just like to express my opinion :-P

**BTW**: If anybody knows some good papers about this topic. Eg benchmarks or bytecode/jit analysis or whatever. Then I would be happy got get a link. As it seems hard to find good material regarding this topic.